prog/dftb+/lib_dftb/h5correction.F90

```diff
@@ -0,0 +1,243 @@
+       write(37,*) "   ", this%speciesName(iSp1), " = ", this%elementPara(iSp1)
+     end do
+  end subroutine printH5Setup
+
+  !> Get H5 parameters for a pair of species
+  subroutine getParams(this, iSp1, iSp2, applyCorrection, h5Scaling, sumVDW)
+    !> Returns a flag whether the correction is applied to the pair,
+    !> and the species-specific parameters
+
+    ! Arguments
+    class(H5Corr), intent(in) :: this
+    !> Input: specie names
+    integer, intent(in) :: iSp1, iSp2
+    !> Output: flag indicating if the correction is applied to that pair
+    logical, intent(out) :: applyCorrection
+    !> Output: pair-specific scaling factor
+    real(dp), intent(out) :: h5Scaling
+    !> Output: Sum of vdW radii of the pair
+    real(dp), intent(out) :: sumVDW
+
+    ! Local variables
+    character(mc) :: spName1, spName2
+    integer :: iSpHeavy
+    character(mc) :: spNameHeavy
+
+    spName1 = this%speciesName(iSp1)
+    spName2 = this%speciesName(iSp2)
+    applyCorrection = .false.
+
+    ! If there is one hydrogen and one other atom, save the heavy atom,
+    ! otherwise return with applyCorrection = false
+    if (spName1 .ne. "H" .and. spName2 == "H") then
+      iSpHeavy = iSp1
+      spNameHeavy = spName1
+    else if (spName2 .ne. "H" .and. spName1 == "H") then
+      iSpHeavy = iSp2
+      spNameHeavy = spName2
+    else
+      return
+    end if
+   
+    ! For each species the correction is applied to,
+    ! the correction is enabled and corresponding
+    ! parameters are returned
+    if (spNameHeavy == "O" ) then
+      ! Correction for OH
+      applyCorrection = .true.
+      ! Parameters
+      h5Scaling = this%elementPara(iSpHeavy)
+      sumVDW = 2.72_dp
+      return
+    end if
+
+    if (spNameHeavy == "N" ) then
+      ! Correction for NH
+      applyCorrection = .true.
+      ! Parameters
+      h5Scaling = this%elementPara(iSpHeavy)
+      sumVDW = 2.75_dp
+      return
+    end if
+
+    if (spNameHeavy == "S" ) then
+      ! Correction for SH
+      applyCorrection = .true.
+      ! Parameters
+      h5Scaling = this%elementPara(iSpHeavy)
+      sumVDW = 3.00_dp
+      return
+    end if
+  end subroutine getParams
+
+  !> Apply the correction to the short-range part of gamma function
+  subroutine scaleShortGamma(this, shortGamma, iSp1, iSp2, rab)
+    !> Returns modified shortGamma
+
+    ! Arguments
+    class(H5Corr), intent(in) :: this
+    real(dp), intent(inout) :: shortGamma
+    integer, intent(in) :: iSp1, iSp2
+    real(dp), intent(in) :: rab
+    
+    ! Local variables
+    real(dp) :: h5Scaling, gauss, sumVDW, fwhm, r0, c
+    logical :: applyCorrection
+
+    ! Get parameters for current pair of species
+    call this%getParams(iSp1, iSp2, applyCorrection, h5Scaling, sumVDW)
+
+    ! If applicable to the current pair, modify the gamma
+    if (applyCorrection) then
+      ! Gaussian calculation
+      fwhm = this%wScale * sumVDW
+      r0 = this%rScale * sumVDW
+      c = fwhm / 2.35482_dp
```

Comment:
Where does the scaling factor come from? Not obvious how this is related to the s_r values in the paper (in the text that is 0.75, in table 1 its 0.714).